src/appServiceNodes.ts

```diff
@@ -123,6 +140,182 @@ export class AppServiceNode extends NodeBase {
     }
 }
 
+type kuduFile = { mime: string, name: string, path: string }; 
+
+export class FilesNode extends NodeBase {
+    constructor(readonly label: string, readonly path: string, readonly site: WebSiteModels.Site, readonly subscription: SubscriptionModels.Subscription) {
+        super(label);
+    }
+
+    getTreeItem(): TreeItem {
+        return {
+            label: this.label,
+            collapsibleState: TreeItemCollapsibleState.Collapsed,
+            iconPath: { 
+                light: path.join(__filename, '..', '..', '..', 'resources', 'light', 'Folder_16x_vscode.svg'),
+                dark: path.join(__filename, '..', '..', '..', 'resources', 'dark', 'Folder_16x_vscode.svg')
+            }
+        }
+    }
+
+    async getChildren(azureAccount: AzureAccountWrapper): Promise<NodeBase[]> {
+        let nodes = [];
+        let user = await util.getWebAppPublishCredential(azureAccount, this.subscription, this.site);
+        let kuduClient = kuduApi({
+            website: this.site.name,
+            username: user.publishingUserName,
+            password: user.publishingPassword
+        });
+        
+        let files : kuduFile[] = await new Promise<kuduFile[]>((resolve, reject) => {
```

Comment:
What's the intention with defining a new promise and then awaiting it? Seems like you could call the code inside the promise directly